packet/pcap_utils.go

```diff
@@ -50,16 +50,22 @@ func WritePcapGlobalHdr(f io.Writer) {
 	}
 	err := binary.Write(f, binary.LittleEndian, &glHdr)
 	if err != nil {
-		common.LogError(common.Debug, "WritePcapGlobalHdr:", err)
+		common.LogErrorNoExit(common.Debug, "WritePcapGlobalHdr:", err)
+		return err
 	}
+	return nil
 }
 
 // WritePcapOnePacket writes one packet with pcap header in file.
 // Assumes global pcap header is already present in file. Packet timestamps have nanosecond resolution.
-func (pkt *Packet) WritePcapOnePacket(f io.Writer) {
+func (pkt *Packet) WritePcapOnePacket(f io.Writer) error {
 	bytes := low.GetRawPacketBytesMbuf(pkt.CMbuf)
-	writePcapRecHdr(f, bytes)
-	writePacketBytes(f, bytes)
+	err := writePcapRecHdr(f, bytes)
+	if err != nil {
+		return err
+	}
+	err = writePacketBytes(f, bytes)
```

Comment:
Hmm.. maybe leaf error assignments should be turned into error returning statements?
```go
// Before:
err = writePacketBytes(f, bytes)
return err
// After:
return writePacketBytes(f, bytes)
```